test/src/test_motion_prediction_model_differential_drive.cpp

```diff
@@ -0,0 +1,226 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <gtest/gtest.h>
+
+#include <mcl_3dl/motion_prediction_models/motion_prediction_model_differential_drive.h>
+
+static const double EPS = 1.0e-6;
+
+TEST(TestMotionPredictionModelDifferentialDrive, predictWithoutNoise)
+{
+  const mcl_3dl::Vec3 relative_trans(0.5, -0.1, 0.1);
+  const mcl_3dl::Quat relative_quat(mcl_3dl::Vec3(0.1, -0.1, 0.3));
+
+  const mcl_3dl::State6DOF odom_prev(mcl_3dl::Vec3(1.0, 1.1, 1.2), mcl_3dl::Vec3(0.3, 0.2, 0.1));
+  const mcl_3dl::State6DOF odom_current(odom_prev.pos_ + odom_prev.rot_ * relative_trans,
+                                        odom_prev.rot_ * relative_quat);
+
+  const mcl_3dl::Vec3 state_trans(1.5, 1.2, 1.3);
+  const mcl_3dl::Quat state_quat(mcl_3dl::Vec3(-0.3, -0.2, -0.1));
+  mcl_3dl::State6DOF state(state_trans, state_quat);
+  state.noise_ll_ = 0.0;
+  state.noise_la_ = 0.0;
+  state.noise_al_ = 0.0;
+  state.noise_aa_ = 0.0;
+
+  mcl_3dl::MotionPredictionModelDifferentialDrive predictor(10.0, 10.0);
+  predictor.setOdoms(odom_prev, odom_current, 1.0);
+  predictor.predict(state);
+
+  const auto expected_state_pos = state_trans + state_quat * relative_trans;
+  EXPECT_NEAR(state.pos_.x_, expected_state_pos.x_, EPS);
+  EXPECT_NEAR(state.pos_.y_, expected_state_pos.y_, EPS);
+  EXPECT_NEAR(state.pos_.z_, expected_state_pos.z_, EPS);
+
+  const mcl_3dl::Quat expected_state_rot = state_quat * relative_quat;
+  EXPECT_NEAR(state.rot_.x_, expected_state_rot.x_, EPS);
+  EXPECT_NEAR(state.rot_.y_, expected_state_rot.y_, EPS);
+  EXPECT_NEAR(state.rot_.z_, expected_state_rot.z_, EPS);
+  EXPECT_NEAR(state.rot_.w_, expected_state_rot.w_, EPS);
+
+  // odom_err_integ_lin_ and odom_err_integ_ang_ are zero vectors.
+  EXPECT_NEAR(state.odom_err_integ_lin_.x_, 0.0, EPS);
+  EXPECT_NEAR(state.odom_err_integ_lin_.y_, 0.0, EPS);
+  EXPECT_NEAR(state.odom_err_integ_lin_.z_, 0.0, EPS);
+  EXPECT_NEAR(state.odom_err_integ_ang_.x_, 0.0, EPS);
+  EXPECT_NEAR(state.odom_err_integ_ang_.y_, 0.0, EPS);
+  EXPECT_NEAR(state.odom_err_integ_ang_.z_, 0.0, EPS);
+}
+
+TEST(TestMotionPredictionModelDifferentialDrive, predictWithoutRotation)
+{
+  const mcl_3dl::Vec3 relative_trans(-0.5, 0.1, 0.05);
+  const float relative_trans_norm = relative_trans.norm();
+
+  const mcl_3dl::State6DOF odom_prev(mcl_3dl::Vec3(5.0, 4.0, 3.0), mcl_3dl::Vec3(0.5, 0.4, 0.3));
+  const mcl_3dl::State6DOF odom_current(odom_prev.pos_ + odom_prev.rot_ * relative_trans, odom_prev.rot_);
+
+  const mcl_3dl::Vec3 state_trans(4.0, 3.0, 2.0);
+  const float state_yaw_angle = 0.3;
+  const mcl_3dl::Quat state_quat(mcl_3dl::Vec3(0.0, 0.0, state_yaw_angle));
+  mcl_3dl::State6DOF state(state_trans, state_quat);
+  state.noise_ll_ = 0.1;
+  state.noise_al_ = 0.2;
+  state.noise_la_ = 0.3;
+  state.noise_aa_ = 0.4;
+
+  const float err_integ_multiply = 0.9;
```

Comment:
```suggestion
  const float err_integ_multiply = 1.0 - (1.0 / 10.0);  // single step of the exponential smoother
```
would be better for the readability of the test condition

Reply:
Done